api/src/main/java/org/openmrs/module/ModuleUtil.java

```diff
@@ -571,7 +577,16 @@ public static void expandJar(File fileToExpand, File tmpModuleDir, String name,
 					String entryName = jarEntry.getName();
 					// trim out the name path from the name of the new file
 					if (!keepFullPath && name != null) {
-						entryName = entryName.replaceFirst(name, "");
+						// if 'name' is a file then make sure that only path is trimmed out and not the file name itself 
+						if (name.equals(entryName)) {
+							int lastSlash = name.lastIndexOf('/');
+							if (lastSlash >= 0) {
+								entryName = entryName.replaceFirst(name.substring(0, lastSlash + 1), "");
```

Comment:
Ok, I can revert the change and get rid of failing test. 
Still I would expect such function to work properly with all parameter combinations and not only with those that we can observe in a present code. :)